src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -192,4 +197,50 @@ public DateTimeField getField(Chronology chronology) {
             return new OffsetDateTimeField(new DividedDateTimeField(new OffsetDateTimeField(chronology.monthOfYear(), -1), QuarterOfYear, 3), 1);
         }
     };
+
+    public static class EpochTimeParser implements DateTimeParser {
+
+        private static final Pattern MILLI_SECOND_PRECISION_PATTERN = Pattern.compile("^\\d{1,13}$");
+        private static final Pattern SECOND_PRECISION_PATTERN = Pattern.compile("^\\d{1,10}$");
+
```

Comment:
I think this might be busted, JSON numerical values are _always_ floats, thus you have a timestamp resulting in `IllegalArgumentException[Invalid format: "1.470417092e+09"];`. Any recommended way around this?